File: name-score/src/main/java/com/shrader/namescore/parser/NameParser.java
package com.shrader.namescore.parser;

import java.nio.CharBuffer;
import java.util.List;


public interface NameParser<A> {

    /**
     * Stream a character buffer and separate names by the supplied delimiter
     *
     * @param dataSource
     * @param delimiter
     * @return List<String>
     */
    List<String> parse(CharBuffer dataSource, String delimiter);
}

File: name-score/name-score-shell/src/main/kotlin/com/shrader/namescore/shell/parser/NameParser.kt
package com.shrader.namescore.shell.parser

import java.nio.CharBuffer


interface NameParser<T> {

    /**
     * Stream a character buffer and separate names by the supplied delimiter
     *
     * @param charBuffer
     * @param delimiter
     * @return List<T>
    </T> */
    fun parse(charBuffer: CharBuffer, delimiter: String): List<String>
}
